XMonad/Actions/BindByLayout.hs

```diff
@@ -27,23 +27,25 @@ import XMonad.StackSet as S
 --
 -- You can use this module with the following in your @~\/.xmonad\/xmonad.hs@:
 --
--- >  import XMonad.Actions.PerLayoutKeys
+-- >  import XMonad.Actions.BindByLayout
 --
--- >   ,((0, xK_F2), bindByLayout [("Tall", spawn "rxvt"), ("Mirror Tall", spawn "xeyes"), ("", spawn "xmessage hello")])
+-- >   ,((0, xK_F2), bindBL [("Tall", spawn "rxvt"), ("Mirror Tall", spawn "xeyes"), ("", spawn "xmessage hello")])
 --
 -- For detailed instructions on editing your key bindings, see
 -- "XMonad.Doc.Extending#Editing_key_bindings".
 
 -- | Uses supplied function to decide which action to run depending on current layout name.
-chooseActionByLayout :: (String->X()) -> X()
-chooseActionByLayout f = withWindowSet (f . description . S.layout. S.workspace . S.current)
+chooseAction :: (String->X()) -> X()
+chooseAction f = withWindowSet (f . description . S.layout. S.workspace . S.current)
 
 -- | If current layout is listed, run appropriate action (only the first match counts!)
 -- If it isn't listed, then run default action (marked with empty string, \"\"), or do nothing if default isn't supplied.
-bindByLayout :: [(String, X())] -> X()
-bindByLayout bindings = chooseActionByLayout chooser where
+bindBL :: [(String, X())] -> X()
+bindBL bindings = chooseAction chooser where
```

Comment:
I'm not sure it's wise to change the quite self-explanatory `bindByLayout` to a name where you basically have to know what it means in order to understand it

Reply:
Also, because this module may already be in use, you would want to retain the old name with a `DEPRECATED` pragma.

Reply:
> I'm not sure it's wise to change the quite self-explanatory `bindByLayout` to a name where you basically have to know what it means in order to understand it

I dont think so, This why i dont like 'PerLayoutKeys` `BindByLayout` is the literal defintion. Renamed it `PerLayoutKeys.hs` Updated `CHANGES.md`

Reply:
> I'm not sure it's wise to change the quite self-explanatory `bindByLayout` to a name where you basically have to know what it means in order to understand it
Before and After
`, ("M-h", bindByLayout [("Full", windows W.focusDown), ("ReflectX Full", windows W.focusUp), ("", sendMessage $ Go L)])`
`, ("M-h", bindBL [("Full", windows W.focusDown), ("ReflectX Full", windows W.focusUp), ("", sendMessage $ Go L)])`

Reply:
Indeed, `bindByLayout` is much more readable!

Reply:
Ok then

Reply:
@slotThe Defaults with typo fixed.

Reply:
https://github.com/xmonad/xmonad-contrib/pull/715/files